src/blueprints/tv2/action-factories/tv2-graphics-action-factory.ts

```diff
@@ -154,4 +195,307 @@ export class Tv2GraphicsActionFactory {
+      partId,
+      fullGraphicPieceInterface,
+      graphicsData
+    )
+  }
+
+  private createFullGraphicsActionFromGraphicsData(
+    keepPreviousPartAliveDuration: number,
+    partId: string,
+    pieceInterface: Tv2PieceInterface,
+    graphicsData: Tv2FullscreenGraphicsManifestData
+  ): Tv2PartAction {
+    const partInterface: PartInterface = this.createGraphicsPartInterface({
+      id: partId,
+      name: `Full ${graphicsData.name}`,
+      inTransition: {
+        keepPreviousPartAliveDuration,
+        delayPiecesDuration: 0
+      },
+    })
+
+    return {
+      id: `full_graphics_${this.stringHashConverter.getHashedValue(graphicsData.name)}`,
+      name: `Full Graphics - ${graphicsData.name}`,
+      type: PartActionType.INSERT_PART_AS_NEXT,
+      data: {
+        partInterface: partInterface,
+        pieceInterfaces: [pieceInterface]
+      },
+      metadata: {
+        contentType: Tv2ActionContentType.GRAPHICS
+      }
+    }
+  }
+
+  private createVizFullGraphicsPieceInterface(partId: string, blueprintConfiguration: Tv2BlueprintConfiguration, fullscreenGraphicsData: Tv2FullscreenGraphicsManifestData): Tv2PieceInterface {
+    return this.createGraphicsPieceInterface({
+      id: 'fullGraphicPiece',
+      partId,
+      name: fullscreenGraphicsData.name,
+      preRollDuration: blueprintConfiguration.studio.VizPilotGraphics.PrerollDuration,
+      pieceLifespan: this.findPieceLifespan(blueprintConfiguration, fullscreenGraphicsData.name),
+      layer: Tv2SourceLayer.PILOT_GRAPHICS,
+      content: {
+        fileName: `${PILOT_PREFIX}${fullscreenGraphicsData.vcpId}`,
+        path: `${fullscreenGraphicsData.vcpId}`,
+      },
+      timelineObjects: this.createVizFullPilotTimelineObjects(blueprintConfiguration, fullscreenGraphicsData)
+    })
+  }
+
+  private createGraphicsPartInterface(partInterfaceWithRequiredValues: Pick<PartInterface, 'id' | 'name'> & Partial<PartInterface>): PartInterface {
+    return {
+      disableNextInTransition: false,
+      expectedDuration: 0,
+      inTransition: {
+        keepPreviousPartAliveDuration: 0,
+        delayPiecesDuration: 0
+      },
+      isNext: true,
+      isOnAir: false,
+      isPlanned: false,
+      outTransition: {
+        keepAliveDuration: 0
+      },
+      pieces: [],
+      rank: 0,
+      segmentId: '',
+      isUnsynced: false,
+      ...partInterfaceWithRequiredValues
+    }
+  }
+
+  private findPieceLifespan(blueprintConfiguration: Tv2BlueprintConfiguration, templateName: string): PieceLifespan {
+    const template: GraphicsTemplate | undefined = blueprintConfiguration.showStyle.graphicsTemplates.find(
+      graphic => graphic.vizTemplate ? graphic.vizTemplate.toUpperCase() === templateName.toUpperCase() : false
+    )
+
+    if (
+      !template ||
+      (!template.outType || !template.outType.toString().length) ||
+      (template.outType !== 'S' && template.outType !== 'O')
```

Comment:
This condition is not that intuitive. One way to handle this would be to use an enumeration for the out type, and set a default value, when it is not set. 

The current condition can be rewritten as:
```typescript
    if (!template || !['S', 'O'].includes(template.outType ?? '')) {
      return PieceLifespan.WITHIN_PART
    }
```

If an enumeration is used with a default value, then this condition can be removed. And it would remove all the magic strings :)

Reply:
Has been simplifed by use of an enum, which the outType is parsed to.